container/test-impl-base/src/main/java/org/jboss/arquillian/container/test/impl/RunModeUtils.java

```diff
@@ -47,31 +49,64 @@ private RunModeUtils() { }
     * @param testMethod
     * @return
     */
-   public static boolean isRunAsClient(Deployment deployment, Class<?> testClass, Method testMethod)
+   public static boolean isRunAsClient(Deployment deployment, TestClass testClass, Method testMethod)
    {
-      boolean runMethodAsClient = testMethod.isAnnotationPresent(RunAsClient.class);
-      boolean runClassAsClient = testClass.isAnnotationPresent(RunAsClient.class);
-
       boolean runAsClient = true;
-      if(deployment != null)
+      if (deployment != null)
       {
-         runAsClient =  deployment.getDescription().testable() ? false:true;
-         runAsClient =  deployment.isDeployed() ? runAsClient:true;
+         runAsClient = deployment.getDescription().testable() ? false : true;
+         runAsClient = deployment.isDeployed() ? runAsClient : true;
```

Comment:
It's a bit fuzzy for me what is going on (not your changes per se, but rather the whole thing). 

First line is rather straightforward, how about making it simpler  thouhg?
`runAsClient = !deployment.getDescription().testable();`,

But I have difficulties understanding what is the logic of the next statement? If it's deployed then value of the previous flag (`testable`) which is ok, but what does it mean when it's not deployed?